### ISSUE TITLE
Models: Clients

### ISSUE THREAD

**Issue by elthran** [elthran/BookingSystem#6]
Information stored about each client. I'm not sure about how to set this up as you said clients don't create accounts to pay. So I'm not sure where to put this. Is it something the business can create, like they can create client profiles and store information about them? Is it needed? What info would you store about the client? Name, address, appointment history? I assume all these fields would be optional?

**Comment by melissachang9** [comment]
Clients book appointments by first choosing an available date and person to do the service.  They should then be asked to input the following information:

First and Last Name
Phone No
Email
Preferred Method of Contact

After completing that screen, clients should be prompted for payment.

Also need to include consent to contact for promotions, etc., and ask them to complete the other consent forms which will be stored electronically and attached to their client file.  This could be done after the booking is complete.  I don't want to make it too cumbersome before they've booked.

**Comment by elthran** [comment]
When they create a second appointment, how would it auto-populate? Just through cookies saved on their browser or would they type in their email or something and have it load their info? This is one of the few objects in the database I will have to think a lot more about before I can code it.

**Comment by melissachang9** [comment]
It won’t auto populate for the client, it will auto populate for the user. As an example, when a client begins booking an appointment, they input their name, email, phone no, etc. And sometimes they don’t make it to the payment section and they end up calling me to manually book over the phone.  When this happens, their information is already in my system which makes it so much easier for me as I just begin typing their name and their information pops up. Same when my existing clients call a year later to book.

Melissa Chang

**Comment by elthran** [comment]
Ok I will need to think about this. Like if the client makes another appointment but spells their name slightly differently it might create multiple instances. So it would need to be set up through a specific variable (likely their email address). So that the first thing they enter is their email address and then as long as you have that, you can bring up their info. So your business will have a list of clients (anyone who has ever made an appointment) and each client will have a unique email address. And you can type in an email address and it will bring up the client linked to it. Does that sound right?

**Comment by melissachang9** [comment]
Yes that sounds good. The system I have now does create double entries and often times I manually delete them which can be annoying. 

Melissa Chang

**Comment by elthran** [comment]
Ok that sounds good. I can set that up now. If you can lay out exactly how clients fill out their forms here https://github.com/elthran/BookingSystem/issues/8 then I will test this out and see if it all works.
